packages/dds/tree/src/feature-libraries/sequence-change-family/changeset/toDelta.ts

```diff
@@ -202,115 +202,104 @@ function applyOrCollectModifications(
     if (modify.value !== undefined) {
         node.value = modify.value.value;
     }
-    if (modify.fields !== undefined) {
-        const protoFields = node.fields ?? {};
-        const modifyFields = modify.fields;
-        for (const key of Object.keys(modifyFields)) {
-            const brandedKey: FieldKey = brand(key);
-            const outNodes = protoFields[key] ?? fail(ERR_MOD_ON_MISSING_FIELD);
```

Comment:
This was not a safe way to check for fields, and could result in prototype pollution if the field key is `__proto__` and it was missing from this node (it would incorrectly return the prototype here instead of a field missing error). Similar less bad issues would happen for fields named "toString", "hasOwnProperties" etc.

Reply:
The correct way to check for a field is `Object.prototype.hasOwnProperty.call(fieldMap, key)`, which getGenericTreeField now uses (it also had the same bug).